### PR TITLE
added survey controller endpoints

### PR DESCRIPTION
**Pull request by cartothemax**
I based this off of the `ContactsController` and `ContactsRepository` to create a similar `Survey` pieces.  Let me know if this is going to be useful for this project and I can start chasing down more of these similar endpoints.

### CODE REVIEW

**Comment by dlogankeenan** [comment]
Nice Addition!! It looks like CI is failing with some Linting errors.  Want to fix those? We'll also want some tests for the Repository too.

I think in future work we'll leverage an action filter to take care of some of the validation work
https://docs.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/model-validation-in-aspnet-web-api#handling-validation-errors

File: dotnetcorehack.Test/Controllers/SurveysControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+            [Fact]
+            public void ShouldReturnNotFoundIfSurveyDoesNotExist()
+            {
+                var response = this.surveysController.Get(this.expectedId);
+
+                Assert.IsType<NotFoundResult>(response);
+            }
+
+            private List<T> N<T>(int count, Func<T> creator)
```
Comment (alexashley) [inline dotnetcorehack.Test/Controllers/SurveysControllerTest.cs]: I know this was copypasta from the other one, but it's possible to ditch the for loop using `Enumerable.Range` 

```cs
Enumerable.Range(0, count).Select(i => creator());

**Comment by dlogankeenan** [comment]
`dotnet build -c release` will build and run linting. I need to add that to the readme

**Comment by cartothemax** [comment]
Sorry about the failing build.  I will get that passing quick here and also refactor that loop out.

Thanks for the feedback!

**Comment by dlogankeenan** [comment]
@cartothemax do you want to remove the repository code? It's isn't tested and we'll take a different direction compared to the contacts example. That used an entity framework in memory database. We'll likely use a postgres database with docker.

**Comment by dlogankeenan** [comment]
@cartothemax I'll remove the repository code. I need to work in the survey controller today. thanks for the contribution 🎉

**Pull request merged**